encryption.go

```diff
@@ -0,0 +1,260 @@
+package memberlist
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/subtle"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/crypto/curve25519"
```

Comment:
Could you use [`crypto/ed25519`](https://golang.org/pkg/crypto/ed25519/) from the standard library for its types and key generation? It's available as of 1.13.

Reply:
No because ed25519 is only for signing, not Diffie-Hellman.

Reply:
Ah! I had to google the difference between X25519 and ed25519 keys in encryption. It's described pretty well in [this blog](https://blog.filippo.io/using-ed25519-keys-for-encryption/). While it's technically possible to convert the keys back and forth, the X25519 keys use a different type of curve that permits constant time operations, which is what we would want here. Got it! 👍

Reply:
Yeah, you can but that's more about converting the data. The algorithm for performing the operations on the data is also different.

Reply:
I understand that EdDSA is not ECDH. I got confused in thinking the keys were essentially the same and used the same curve. That was wrong. 😅